src/ImpurityDensities/ThermalDiffusionLithiumDensity.jl

```diff
@@ -0,0 +1,66 @@
+"""
+    struct ThermalDiffusionLithiumDensity{T <: SSDFloat} <: AbstractImpurityDensity{T}
+
+Lithium impurity density model
+
+ref: [Dai _et al._ (2023)](https://doi.org/10.1016/j.apradiso.2022.110638)
+ 
+## Fields
+* `lithium_annealing_temperature::T`: lithium annealing temperature in Kelvin, when the lithium is diffused into the crystal. The default value is 623 K.
+* `lithium_annealing_time::T`: lithium annealing time in seconds. The default value is 18 minutes.
+* `distance_to_contact::Function`: the function for describing the depth to surface. 
+    1) use the built-in function `ConstructiveSolidGeometry.distance_to_surface` to calculate the distance to the surface of the doped contact 
+    2) custom. 
+    Custom function might be much faster while the detector has good symmetry.
+* `lithium_density_on_contact::T`: the lithium concentration in the surface (in m^-3)
+* `lithium_diffusivity_in_germanium::T`: the diffusivity of lithium in germanium (in m^2*s^-1)
+"""
+
+struct ThermalDiffusionLithiumDensity{T <: SSDFloat, G <: Union{<:AbstractGeometry, Nothing}} <: AbstractImpurityDensity{T}
+	lithium_annealing_temperature::T
+	lithium_annealing_time::T
+    contact_with_lithium_doped::G
+    distance_to_contact::Function
+	lithium_density_on_contact::T
+    lithium_diffusivity_in_germanium::T
+end
```

Comment:
The indentation is a bit odd here.
```suggestion
struct ThermalDiffusionLithiumDensity{T <: SSDFloat, G <: Union{<:AbstractGeometry, Nothing}} <: AbstractImpurityDensity{T}
    lithium_annealing_temperature::T
    lithium_annealing_time::T
    contact_with_lithium_doped::G
    distance_to_contact::Function
    lithium_density_on_contact::T
    lithium_diffusivity_in_germanium::T
end
```
Also, we might not need to save the `contact_geometry` here, if it is already saved in the `distance_to_contact` function. Not sure how this affects the runtime performance-wise, though.